src/main/java/seedu/address/storage/Authentication.java

```diff
@@ -0,0 +1,202 @@
+    }
+
+    /**
+     * Initiates a feedback loop to get the user's password to unlock the encrypted zip file.
+     */
+    public void unlock() throws ZipException {
+        assert isExistsZip() : "Zip must exist to call the unlock() method.";
+
+        //If there is zip but it's not locked, just unzip it.
+        if (!isExistsLockedZip()) {
+            attemptUnzip();
+        } else {
+            Scanner scanner = new Scanner(System.in);
+            while (true) {
+                System.out.println("Please enter your password: ");
+                String attemptPassword = scanner.nextLine();
+                if (attemptUnzip(attemptPassword)) {
+                    this.password = Optional.of(attemptPassword);
+                    break;
+                } else {
+                    System.out.println("Password is incorrect, please try again.");
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if the locked zip containing the data file exists.
+     * @return true if the locked zip exists, false otherwise.
+     */
+    public boolean isExistsZip() {
+        return Files.exists(Paths.get(this.getZipPath()));
+    }
+
+    /**
+     * Checks if the locked zip file containing the data .json file exists.
+     * @return true if the zip file exists.
+     * @throws ZipException if error occurs when checking if the zip file is encrypted.
+     */
+    public boolean isExistsLockedZip() throws ZipException {
+        ZipFile dataZip = new ZipFile(this.getZipPath());
+        return isExistsZip() && dataZip.isEncrypted();
+    }
+
+    /**
+     * Locks the data .json file in a password protected zip file and delete the original data .json file.
+     * Assumes: data .json file exists.
+     */
+    public void lock() throws ZipException {
+        ZipFile dataZip;
+        try {
+            //If ClientBook is locked, encrypt the zip file using the password.
+            if (this.password.isPresent()) {
+                dataZip = new ZipFile(this.getZipPath(), this.password.get().toCharArray());
+                ZipParameters parameters = new ZipParameters();
+                parameters.setEncryptFiles(true);
+                parameters.setEncryptionMethod(EncryptionMethod.AES);
+                parameters.setAesKeyStrength(AesKeyStrength.KEY_STRENGTH_256);
+                parameters.setCompressionLevel(CompressionLevel.NORMAL);
+                parameters.setCompressionMethod(CompressionMethod.DEFLATE);
+                dataZip.addFile(this.filePath.toString(), parameters);
+
+            //ClientBook is not locked, zip the data .json file without a password.
+            } else {
+                dataZip = new ZipFile(this.getZipPath());
+                dataZip.addFile(this.filePath.toString());
+            }
+        } catch (ZipException e) {
+            logger.info("Failed to compress data .json file into a zip file");
+            throw e;
+        }
+
+
+
+
```

Comment:
Just a little bit of unnecessary white space :)

Reply:
Fixed.